oe_subscriptions.tokens.inc

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * @file
+ * Token callbacks for the Subscriptions module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_token_info().
+ */
+function oe_subscriptions_token_info(): array {
+  $info = [];
+
+  $info['tokens']['user']['subscriptions-page'] = [
+    'name' => t('Subscriptions page URL'),
+    'description' => t('The URL to subscriptions page.'),
+    'type' => 'url',
+  ];
+
+  return $info;
+}
+
+/**
+ * Implements hook_tokens().
+ */
+function oe_subscriptions_tokens(string $type, array $tokens, array $data, array $options, BubbleableMetadata $bubbleable_metadata): array {
+  $replacements = [];
+
+  if (
+    $type === 'user' &&
+    !empty($data['user'])
+  ) {
+    // Adding the URL to the subscriptions page will lead into an 403 for
+    // non-logged users, we add a login URL with the subscriptions page
+    // destination.
+    $url = Url::fromRoute('user.login', [], [
+      'query' => [
+        'destination' => Url::fromRoute('oe_subscriptions.user.subscriptions', [
+          'user' => $data['user']->id(),
+        ])->toString(),
+      ],
+    ]);
```

Comment:
This looked good at a first glance, but now testing the prefixed token generation, I've noticed that using a token like `subscriptions-page:path` will output `/user/login` which is incorrect.
After talking with @AaronGilMartinez , we agreed that preventing the user to hit the 403 is not the scope of the token in this module. It can be either customised in the project, or a module like https://www.drupal.org/project/r4032login can be used to configure that specific url to redirect to the login page.

Reply:
Added changes to replace route